draft-ietf-jsonpath-base.md

```diff
@@ -958,19 +961,29 @@ In the resultant nodelist:
 
 Children of an object may occur in any order, since JSON objects are unordered.
 
-### Union Selector
+### List Selector
+
+The list selector allows combining member names, array indices, and
+slices in a single selector.
+
+Note: The list selector was called "union selector" in
+{{JSONPath-orig}}, as it was intended to solve use cases addressed by
+the union selector in XPath.
+However, the term "union" implies the set operation that involves
+merging input sets while avoiding duplicates, so the concept was
+renamed into "list selector".
 
 #### Syntax
 {: unnumbered}
 
-The union selector is syntactically related to the
-`index-selector`.
+The list selector is syntactically related to the
+`index-selector` and the `slice-selector`.
 It contains two or more entries, separated by commas.
 
 ~~~~ abnf
-union-selector = "[" S union-entry 1*(S "," S union-entry) S "]"
+list-selector  = "[" S list-entry 1*(S "," S list-entry) S "]"
 
-union-entry    =  ( quoted-member-name /
+list-entry     =  ( quoted-member-name /
                     element-index      /
                     slice-index
                   )
```

Comment:
It seems that lists can't contain filters (neither could unions before this PR). I wonder if this is deliberate or accidental? The comparison project shows there is [no consensus](https://cburgmer.github.io/json-path-comparison/results/union_with_filter.html) among implementations.

Reply:
Interesting. I'm not sure that you should be forbidden to do so, either. I believe I support it already.

Reply:
Now https://github.com/ietf-wg-jsonpath/draft-ietf-jsonpath-base/issues/150